e2e/test/helpers/TestDevice.cs

```diff
@@ -79,7 +79,7 @@ internal string ConnectionString
         /// <summary>
         /// Device identity object.
         /// </summary>
-        internal Device Device { get; private set; }
+        internal Device Device { get; set; }
```

Comment:
Allow a test using `TestDevice` to update the property when it performs an operation that updates the device.